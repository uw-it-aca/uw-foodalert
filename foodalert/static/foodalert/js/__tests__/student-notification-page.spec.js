"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import {mount} from '@vue/test-utils';
import NotificationPage from '../pages/student/notification-page.vue';

describe('Notification Page', function () {
        test('renders correctly', function () {
            var wrapper = mount(NotificationPage, {
            });
            expect(wrapper.element).toMatchSnapshot();
        });
})