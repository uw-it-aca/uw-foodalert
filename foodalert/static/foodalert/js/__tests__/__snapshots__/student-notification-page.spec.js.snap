// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Page renders correctly 1`] = `
<div
  class="page"
>
  <div
    id="standard-notification"
  >
    <div
      class="collapse"
      id="notif-container"
      style="display: none;"
    >
      <div
        class="text-center py-3 container-fluid"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-8"
            >
              <div
                class="h5 text-white pt-2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="page-content pb-2"
  >
     
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="page-content-padding col-md-8"
        >
          <h1
            class="pt-4 pb-2"
            id="standard-heading"
          >
            
        Notification preferences
    
          </h1>
           
          <div
            id="standard-body"
          >
            <p>
              
            Select how you will like to receive notifications. Please choose at least one.                                                      
        
            </p>
             
            <div
              class="notification"
            >
              <div
                class="collapse"
                id="enable-notification"
                style="display: none;"
              >
                <div
                  class="container"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-sm-9 col-9"
                    >
                      <strong>
                        Enable notifications
                      </strong>
                       
                      <p>
                        Turn on to receive notifications. Please choose at least one.
                      </p>
                       
                      <div
                        id="notif-status"
                      >
                        <div
                          class="paused"
                        >
                           Notifications are paused 
                        </div>
                      </div>
                    </div>
                     
                    <div
                      class="col-sm-3 col-3 align-self-center"
                    >
                      <div
                        class="float-right custom-control custom-switch"
                      >
                        <input
                          autocomplete="off"
                          class="custom-control-input"
                          name="enable-switch"
                          type="checkbox"
                          value="true"
                        />
                        <label
                          class="custom-control-label"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="card notif-option-card"
            >
              <!---->
              <div
                class="card-header"
              >
                <div
                  class="p-0 container"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-8"
                    >
                      
                Text 
             
                      <!---->
                       
                      <!---->
                    </div>
                     
                    <div
                      class="col-4"
                    >
                      <div>
                        <a
                          aria-controls="text"
                          aria-expanded="false"
                          class="btn opt_link_btn p-0 btn-link btn-block"
                          href="#"
                          role="button"
                          target="_self"
                        >
                          <span
                            aria-hidden="true"
                            class="mr-2 spinner-padding spinner-border spinner-border-sm spinner-hide"
                          >
                            <!---->
                          </span>
                           Cancel
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="collapse show"
                id="text"
                role="tabpanel"
              >
                <div
                  class="card-body"
                >
                  <!---->
                  <!---->
                  <div
                    class="p-0 container"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="col-12"
                      >
                        <p
                          class="card-text"
                        >
                          <form
                            class=""
                          >
                            <small
                              class="form-text text-muted pt-0"
                            >
                              Enter a new phone number
                            </small>
                             
                            <fieldset
                              class="form-group"
                            >
                              <!---->
                              <div
                                role="group"
                                tabindex="-1"
                              >
                                <input
                                  aria-required="true"
                                  class="form-control"
                                  required="required"
                                  type="text"
                                  width="300px"
                                />
                                <!---->
                                <!---->
                                <small
                                  class="form-text text-muted"
                                  tabindex="-1"
                                >
                                  Carrier rates may apply
                                </small>
                              </div>
                            </fieldset>
                             
                            <!---->
                             
                            <button
                              class="btn float-right mt-2 px-3 btn-primary"
                              type="submit"
                            >
                              <span
                                aria-hidden="true"
                                class="mr-2 spinner-padding spinner-border spinner-border-sm spinner-hide"
                              >
                                <!---->
                              </span>
                              
                                    Verify
                                
                            </button>
                          </form>
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card notif-option-card"
            >
              <!---->
              <div
                class="card-header"
              >
                <div
                  class="p-0 container"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-8"
                    >
                      
                Email 
             
                      <!---->
                       
                      <!---->
                    </div>
                     
                    <div
                      class="col-4"
                    >
                      <div>
                        <a
                          aria-controls="email"
                          aria-expanded="false"
                          class="btn opt_link_btn p-0 btn-link btn-block"
                          href="#"
                          role="button"
                          target="_self"
                        >
                          
                        Add
                    
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="collapse"
                id="email"
                role="tabpanel"
                style="display: none;"
              >
                <div
                  class="card-body"
                >
                  <!---->
                  <!---->
                  <div
                    class="p-0 container"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="col-12"
                      >
                        <p
                          class="card-text"
                        >
                          <form
                            class=""
                          >
                            <small
                              class="form-text text-muted pt-0"
                            >
                              Enter an email
                            </small>
                             
                            <fieldset
                              class="form-group"
                            >
                              <!---->
                              <div
                                role="group"
                                tabindex="-1"
                              >
                                <input
                                  aria-required="true"
                                  class="form-control"
                                  required="required"
                                  type="email"
                                  width="300px"
                                />
                                <!---->
                                <!---->
                                <small
                                  class="form-text text-muted"
                                  tabindex="-1"
                                >
                                  Placeholder description
                                </small>
                              </div>
                            </fieldset>
                             
                            <!---->
                             
                            <button
                              class="btn float-right mt-2 px-3 btn-primary"
                              type="submit"
                            >
                              <span
                                aria-hidden="true"
                                class="mr-2 spinner-padding spinner-border spinner-border-sm spinner-hide"
                              >
                                <!---->
                              </span>
                              
                                    Verify
                                
                            </button>
                          </form>
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
           
        </div>
      </div>
    </div>
  </div>
   
  <footer
    class="text-center"
    id="relative-footer"
  >
    <a
      href="mailto:help@uw.edu?subject=Hungry Husky support"
    >
      Contact support
    </a>
     
    <p>
      © 2019 University of Washington
    </p>
  </footer>
</div>
`;
